File: Day2/mergesort.cpp
Select tags
0/5
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:

        #mergesort implementation
        def mergesort(arr):

            if len(arr) > 1:

                #get mid element to split array
                mid = len(arr)//2

                #split the array into two
                arr_l = arr[:mid]
                arr_r = arr[mid:]

                #recrusive call to left and right halves
                mergesort(arr_l)
                mergesort(arr_r)

                i = j = k = 0

                #compare, swap and sort the elements
                while i<len(arr_l) and j<len(arr_r):
                    if arr_l[i] < arr_r[j]:
                        arr[k] = arr_l[i]
                        i += 1
                    else:
                        arr[k] = arr_r[j]
                        j += 1
                    k += 1
                
                while i<len(arr_l):
                    arr[k] = arr_l[i]
                    i += 1
                    k += 1
                
                while j<len(arr_r):
                    arr[k] = arr_r[j]
                    j += 1
                    k += 1

            return arr

        op = mergesort(nums)
        return op

File: Day47/mergesort.py

#Merge sort

def mergesort(arr):
    
    if len(arr) > 1:
        
        mid = len(arr)//2
        L = arr[:mid]
        R = arr[mid:]
        
        mergesort(L)
        mergesort(R)
        
        i = j = k = 0
        
        while i < len(L) and j < len(R):
            if L[i] < R[j]:
                arr[k] = L[i]
                i = i + 1
            else:
                arr[k] = R[j]
                j = j + 1
                
            k = k + 1
        
        while i < len(L):
            arr[k] = L[i]
            i = i + 1
            k = k + 1
            
        while j < len(R):
            arr[k] = R[j]
            j = j + 1
            k = k + 1

arr = [34, 11, 5, 6, 9]
mergesort(arr)
print(arr)
